### DISCUSSION TITLE
Data Metrics cleanup (Prometheus windows Version)

### DISCUSSION THREAD

**Discussion by AboSalah-1982** [prometheus/prometheus#12786] [Q&A] [answered]
### Proposal

how to do cleanup for old data for Prometheus in windows version?

at the minute it seems to be an infinitely growing data store with no way to clean old data

**Comment by roidelapluie** [comment, marked as answer]
To set a time-based data retention policy in Prometheus on Windows, use the --storage.tsdb.retention.time flag:

```
prometheus.exe --storage.tsdb.retention.time=5d
```

To set a size-based retention policy, use the --storage.tsdb.retention.size flag:

```
prometheus.exe --storage.tsdb.retention.size=5GB
```

You can combine both flags to enforce both time and size limits:

```
prometheus.exe --storage.tsdb.retention.time=5d --storage.tsdb.retention.size=5GB
```

Update these flags in your service configuration if you're running Prometheus as a service, then restart it.
